File: Data Structure/Hash/Medium/LET_no1198_Find Smallest Common Element in All Rows.py
class Solution:
    def smallestCommonElement(self, mat: List[List[int]]) -> int:
        m, n = len(mat), len(mat[0])
        memo = collections.defaultdict(int)
        for row in mat:
            for num in row:
                memo[num] += 1
                if memo[num] == m:
                    return num
        
        return -1

File: Data Structure/Hash/Medium/LET_no1198_Find Smallest Common Element in All Rows.cpp
class Solution {
public:
    int smallestCommonElement(vector<vector<int>>& mat) {
        vector<int> memo(10001, 0);
        int m = mat.size();
        for (auto row : mat) {
            for (int num : row) {
                memo[num]++;
                if (memo[num] == m)
                    return num;
            }
        }
        
        return -1;
    }
};